pkg/querier/distributor_queryable.go

```diff
@@ -184,12 +185,11 @@ func (q *distributorQuerier) streamingSelect(ctx context.Context, sortSeries boo
 			return storage.ErrSeriesSet(err)
 		}
 
-		serieses = append(serieses, &chunkSeries{
-			labels:            ls,
-			chunks:            chunks,
-			chunkIteratorFunc: q.chunkIterFn,
-			mint:              minT,
-			maxt:              maxT,
+		serieses = append(serieses, &storage.SeriesEntry{
+			Lset: ls,
+			SampleIteratorFn: func(iterator chunkenc.Iterator) chunkenc.Iterator {
+				return q.chunkIterFn(chunks, model.Time(minT), model.Time(maxT))
```

Comment:
I am also thinking about getting rid of this `chunkIterFn` completely so that we can remove a ton of code and just reuse upstream interfaces. It makes so much easier to support native histograms.

But this breaks the batch iterator we have.

```
		iterables := make([]chunkenc.Iterable, len(chunks))
		serieses = append(serieses, &storage.SeriesEntry{
			Lset: ls,
			SampleIteratorFn: func(iterator chunkenc.Iterator) chunkenc.Iterator {
				for i, c := range chunks {
					iterables[i] = c.Data.SampleIterable()
				}
				return storage.ChainSampleIteratorFromIterables(iterator, iterables)
			},
		})
```